### ISSUE TITLE
Error bootstrap.php.cache after 3.2 upgrade

### ISSUE THREAD

**Issue by Herod** [symfony/symfony#20753]
HI,

I have updated to 3.2 but I get the following error:

 `PHP Fatal error:  Cannot redeclare class Symfony\\Component\\HttpKernel\\HttpKernelInterface in /var/www/application/var/bootstrap.php.cache on line 1309
`

I cleared the cache but without success.
When I removed the bootstrap.php.cache file everything worked as expected. After the composer update the bootstrap file was regenerated with the same error being present.

line 1309 to 1339:

```
interface HttpKernelInterface
{
const MASTER_REQUEST = 1;
const SUB_REQUEST = 2;
public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true);
}
}
namespace Symfony\Component\HttpKernel
{
use Symfony\Component\DependencyInjection\ContainerInterface;
use Symfony\Component\HttpKernel\Bundle\BundleInterface;
use Symfony\Component\Config\Loader\LoaderInterface;
interface KernelInterface extends HttpKernelInterface, \Serializable
{
public function registerBundles();
public function registerContainerConfiguration(LoaderInterface $loader);
public function boot();
public function shutdown();
public function getBundles();
public function getBundle($name, $first = true);
public function locateResource($name, $dir = null, $first = true);
public function getName();
public function getEnvironment();
public function isDebug();
public function getRootDir();
public function getContainer();
public function getStartTime();
public function getCacheDir();
public function getLogDir();
public function getCharset();
}
```

**Comment by nicolas-grekas** [comment]
Do you call addClassesToCompile in your bundle? Did you put AppKernel in the list? Then that may be the issue.

**Comment by Herod** [comment]
I don't think i know about the addClassesToCompile and where this should be put.

**Comment by rodush** [comment]
Hi! I've gotten similar problem with one of my applications, so I was wondering what makes it different from another, which worked perfectly fine after upgrade to 3.2
I took a look at (symfony demo application's composer.json)[https://github.com/symfony/symfony-demo/blob/master/composer.json]
... and found that in my composer.json I had these lines:
```
-        "psr-0": {
-            "": "src/"
-        },
-        "files": ["app/AppKernel.php"]
+        "psr-4": {"": "src/"},
+        "classmap": ["app/AppKernel.php", "app/AppCache.php"]
```
so after I changed `files` to `classmap` it started working fine.

**Comment by lisachenko** [comment]
This could be solved by #20668 )
